src/teleport.cpp

```diff
@@ -0,0 +1,93 @@
+#include "teleport.h"
+
+#include "character.h"
+#include "avatar.h"
+#include "creature.h"
+#include "player.h"
+#include "monster.h"
+#include "game.h"
+#include "map.h"
+#include "messages.h"
+#include "point.h"
+
+
+const efftype_id effect_teleglow( "teleglow" );
+
+bool teleport::teleport( Creature *c, int min_distance, int max_distance, bool safe,
+                         bool add_teleglow )
+{
+    if( c == nullptr || min_distance > max_distance ) {
+        debugmsg( "ERROR: Function teleport::teleport called with invalid arguments." );
+        return false;
+    }
+
+    bool c_is_u = ( c == &g->u );
+    player *p = dynamic_cast<player *>( c );
```

Comment:
you should use as_player() here instead of dynamic_cast

Reply:
Thanks! Didnt know about that one.